core/src/main/scala/kafka/log/Log.scala

```diff
@@ -335,13 +336,20 @@ class Log(@volatile var dir: File,
   /** The name of this log */
   def name  = dir.getName()
 
-  def leaderEpochCache: LeaderEpochFileCache = _leaderEpochCache
+  def recordVersion: RecordVersion = config.messageFormatVersion.recordVersion
 
-  private def initializeLeaderEpochCache(): LeaderEpochFileCache = {
-    // create the log directory if it doesn't exist
-    Files.createDirectories(dir.toPath)
-    val checkpointFile = new LeaderEpochCheckpointFile(LeaderEpochFile.newFile(dir), logDirFailureChannel)
-    new LeaderEpochFileCache(topicPartition, logEndOffset _, checkpointFile)
+  private def initializeLeaderEpochCache(): Option[LeaderEpochFileCache] = {
+    val leaderEpochFile = LeaderEpochFile.newFile(dir)
+    if (recordVersion.precedes(RecordVersion.V2)) {
+      // Delete the checkpoint file if it exists and the message format is does not support it
```

Comment:
nit: `and the message format is does not support it` => `as the message format does not support it` ?